pkg/image/importer/importer.go

```diff
@@ -537,6 +538,28 @@ func (isi *ImageStreamImporter) importRepositoryFromDocker(ctx gocontext.Context
 			manifest = m
 		}
 
+		if manifestList, isManifestList := manifest.(*manifestlist.DeserializedManifestList); isManifestList {
+			manifest = nil
+			for _, m := range manifestList.Manifests {
+				if m.Platform.OS == "linux" && m.Platform.Architecture == "amd64" {
+					manifest, err = s.Get(ctx, m.Digest)
+					if err != nil {
+						glog.V(5).Infof("unable to access digest %q for tag %q for repository %#v: %#v", m.Digest, importTag.Name, repository, err)
+						importTag.Err = formatRepositoryError(repository, importTag.Name, "", err)
+						continue
+					}
+				}
+			}
+			if manifest == nil {
+				manifest, err = s.Get(ctx, manifestList.Manifests[0].Digest)
+				if err != nil {
+					glog.V(5).Infof("unable to access digest %q for tag %q for repository %#v: %#v", manifestList.Manifests[0].Digest, importTag.Name, repository, err)
+					importTag.Err = formatRepositoryError(repository, importTag.Name, "", err)
+					continue
+				}
+			}
+		}
```

Comment:
Why is this not handled in for i := range repository.Digests { loop as well?

Reply:
Because it supposed to be a hack to unblock the queue.